### PR TITLE
Feat/cancel pending transaction

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 7f46856 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9449452825), [Test report](https://leather-wallet.github.io/playwright-reports/feat-cancel-pending-transaction), [Storybook](https://feat-cancel-pending-transaction--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat-cancel-pending-transaction)

Copy of https://github.com/leather-wallet/extension/pull/5501

## Summary by CodeRabbit

- **New Features**
  - Introduced a cancel transaction button allowing users to cancel transactions.
  - Added a new dialog for canceling Stacks transactions with detailed transaction information and actions.
  - Implemented a new route for the cancel transaction dialog.

- **Enhancements**
  - Updated the transaction item layout to include a new action button group element.
  - Modified the increase fee button to conditionally render based on the active state.

- **Refactor**
  - Renamed `rightElement` prop to `actionButtonGroupElement` across various components for consistency.

### CODE REVIEW

**Comment by alter-eggo** [comment]
I tested this feature and seems works ok
https://explorer.hiro.so/txid/0x8f38b87c9d1e9f60bc63b00aa82a95636529508f11da26ddb5d3eb584653c2ee?chain=mainnet
https://explorer.hiro.so/txid/0x2c31b9cd82d90e4857fa6280a3ff8f74a9d8e67b7ebc5eb93381d86415e1ac2e?chain=mainnet

@314159265359879 could you please test as well

File: src/app/features/dialogs/cancel-transaction-dialog/hooks/use-stx-cancel-transaction.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { useCurrentStacksNetworkState } from '@app/store/networks/networks.hooks';
+import { useSubmittedTransactionsActions } from '@app/store/submitted-transactions/submitted-transactions.hooks';
+import { useRawDeserializedTxState, useRawTxIdState } from '@app/store/transactions/raw.hooks';
+
+import { useSelectedTx } from './use-selected-tx';
+
+export function useStxCancelTransaction() {
+  const [rawTxId, setRawTxId] = useRawTxIdState();
+  const tx = useSelectedTx();
+  const [, setTxId] = useRawTxIdState();
```
Comment (fbwoolf) [inline src/app/features/dialogs/cancel-transaction-dialog/hooks/use-stx-cancel-transaction.ts:28]: I am going to block this PR from getting merged. So much of this code is just duplicated w/out evaluating if needed or even working correctly. I have had to refactor quite a bit of the increase fee dialog changes here: https://github.com/leather-wallet/extension/pull/5500

See here where I'm making this comment, there are two uses of `useRawTxIdState()` in this file to set the id. This component is a mess and caused an infinite loop and broke the wallet in the linked PR.

**Review by fbwoolf — CHANGES_REQUESTED** [review]
This work needs to be refactored so we aren't just duplicating broken code.

File: src/app/components/stacks-transaction-item/cancel-transaction-button.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        zIndex={999}
+      >
+        <HStack gap="space.01">
+          <styled.span textStyle="label.03" color="yellow.action-primary-default">
+            Cancel transaction
+          </styled.span>
+        </HStack>
+      </styled.button>
+    )
+  );
```
Comment (edgarkhanzadian) [inline src/app/components/stacks-transaction-item/cancel-transaction-button.tsx:37]: nit: I'd suggest short circuiting the component's render with an if statement
```suggestion
  if(!isActive) {
    return null;
  }
  return (
      <styled.button
        _hover={{ color: 'ink.text-subdued' }}
        bg="ink.background-primary"
        maxWidth="125px"
        ml="auto"
        onClick={e => {
          onCancelTransaction();
          e.stopPropagation();
        }}
        pointerEvents={!isActive ? 'none' : 'all'}
        position="relative"
        px="space.02"
        py="space.01"
        rounded="xs"
        zIndex={999}
      >
        <HStack gap="space.01">
          <styled.span textStyle="label.03" color="yellow.action-primary-default">
            Cancel transaction
          </styled.span>
        </HStack>
      </styled.button>
  );
```

File: src/app/components/stacks-transaction-item/increase-fee-button.tsx [hunk 3]
```diff
@@ -12,27 +12,28 @@ export function IncreaseFeeButton(props: IncreaseFeeButtonProps) {
...
-      py="space.01"
-      rounded="xs"
-      zIndex={999}
-    >
-      <HStack gap="space.01">
-        <ChevronsRightIcon color="stacks" variant="small" />
-        <styled.span textStyle="label.03">Increase fee</styled.span>
-      </HStack>
-    </styled.button>
+    isActive && (
```
Comment (edgarkhanzadian) [inline src/app/components/stacks-transaction-item/increase-fee-button.tsx:15]: here too

**Review by edgarkhanzadian — DISMISSED** [review]

**Comment by Aman-zishan** [comment]
@alter-eggo reworked PR opened for review #5533

**Comment by pete-watters** [comment]
I'll close this as a new PR from a different branch name was opened so I set it up [here](https://github.com/leather-wallet/extension/pull/5536)

**Pull request closed without merging**